### ISSUE TITLE
Customize JSON stringification behavior of firestore.GeoPoint

### ISSUE THREAD

**Issue by cola119** [firebase/firebase-js-sdk#3605]
### [REQUIRED] Describe your environment

  * Operating System version: macOS Mojave 10.14.6
  * Browser version: Googe Chrome 84.0.4147.105
  * Firebase SDK version: ^7.11.0
  * Firebase Product: firestore (auth, database, storage, etc)

### [REQUIRED] Describe the problem

When stringify `GeoPoint` instance, it becomes like a `{_lat: 30, _long: 133}`. The private fields have been exposed and so confusing.
For example, in nuxtjs, the data fetched on the server-side is serialized and posted to the client-side. This variable results above and we have to access like `geopoint._lat` instead of `geopoint.latitude`.

So I propose implementing `toJSON` method like below:
```ts
class GeoPoint {
  // ...
  toJSON = () => {
    return {
      longitude: this._long,
      latitude: this._lat
    }
  }
}
```
[toJSON behavior reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior)

What do you think about this proposal?

#### Relevant Code:

[TS Playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDCCuAnRBTAdgFwOIoPYAVcBLTaAbwChpoAHRIgNzAxWgH1wMAuaNeAWwBGKRAG4qtekxbsQuNAHMefISPETg8iBkTxgGXIgAUnIhngATFMoHDEAGmhzFZy9d62RASnITqGAAsiCAA6DmZoAF4nZlcrP2hA4LDnBSineQU4lAkAXwoJBRQMGIxsoy8bVURfampUc0Q0RKDQ8Iw8guoiktTyyo9q2rqGpGakttTO-IpQSBgAMSIADxQLHAJiUkpqOkZmVnaqu3FdqQPZTOO1LuhNNG1dfUMTWPMra4cMl3f3FTsfDs6hMUhFoqZfgkQRxMuk+pDqDNusVSv1PsN6sUxi1ku1OoUUfC3BV0UDMY1xq0Upl8dQAPR0u7wbS4fhEABerEeJCyADMiMBYvIJAYAFIAZQA8gA5dIVKIAPgx0FGTWV1CJHxxk0y9gSGrebh40M4CXyiIoM3u2iZyHQJWiaBQAHc4EhUJgNoQSBgjABmACsjgAjH6ACxeWZaXAgFAhOQKIzAd32nwM8jQABE7UzPEDjmzqVz0FDYeguWgBWtJX5qws6SdrqWda9W19+ZL4cj1pjcYTRlrazTjLIWc1KGLpYLELcxcD5egQA)

```javascript
// TODO(you): code here to reproduce the problem
```

**Comment by var-const** [comment]
Thank you for the feature request. We can add a well-defined JSON representation for `GeoPoint` and `Timestamp` (though not for other classes in the API); this will be fixed in the next release after #3615 is merged.

Also, thanks a lot for a very detailed description!

**Comment by cola119** [comment]
Thanks! It's a big help.
